File: animation/src/main/java/me/wyne/wutils/animation/runnable/ParticleArray.java
package me.wyne.wutils.animation.runnable;

import me.wyne.wutils.animation.AnimationRunnable;
import me.wyne.wutils.animation.data.AnimationParticle;
import org.bukkit.World;
import org.bukkit.util.Vector;
import org.jetbrains.annotations.NotNull;

import java.util.Set;

/**
 * Spawns the same {@code particle} at every point in {@code points} within {@code world}.
 */
public record ParticleArray(@NotNull World world, @NotNull Set<@NotNull Vector> points, @NotNull AnimationParticle particle) implements AnimationRunnable {

    @Override
    public void run() {
        for (Vector point : points) {
            particle.spawnParticle(world, point, false);
        }
    }

}

File: animation/src/main/kotlin/me/wyne/wutils/animation/runnable/ParticleArray.kt
package me.wyne.wutils.animation.runnable

import me.wyne.wutils.animation.AnimationRunnable
import me.wyne.wutils.animation.data.AnimationParticle
import org.bukkit.World
import org.bukkit.util.Vector

class ParticleArray(
    private val world: World,
    private val points: Set<Vector>,
    private val particle: AnimationParticle,
) : AnimationRunnable {

    override fun run() {
        points.forEach { particle.spawnParticle(world, it) }
    }

}